src/test/scala/mesosphere/mesos/TaskBuilderTest.scala

```diff
@@ -853,15 +838,15 @@ class TaskBuilderTest extends MarathonSpec {
   }
 
   private def assertTaskInfo(taskInfo: TaskInfo, taskPorts: Seq[Long], offer: Offer): Unit = {
-    val range = taskInfo.getResourcesList.asScala
+    val maybeRange = taskInfo.getResourcesList.asScala
       .find(r => r.getName == Resource.PORTS)
       .map(r => r.getRanges.getRange(0))
-    assert(range.isDefined)
+    assert(maybeRange.isDefined)
+    val range = scala.collection.immutable.Range.Long.inclusive(maybeRange.get.getBegin, maybeRange.get.getEnd, 1)
     // The taskName is the elements of the path, reversed, and joined by dots
     assert("frontend.product" == taskInfo.getName)
     assert(2 == taskPorts.size)
-    assert(taskPorts.head == range.get.getBegin.toInt)
-    assert(taskPorts(1) == range.get.getEnd.toInt)
+    assert(taskPorts.forall(range.contains(_)))
```

Comment:
Wouldn't it be better to compare the sets of ports? E.g. `taskPorts.toSet == range.toSet`

In theory, it is possible that we end up with multiple ranges but maybe I shouldn't be so picky. If I am right, this should happen to 1 in thousand test runs.